File: sdk-node/examples/invoices/update.ts
/**
 * Revise a draft invoice. Only legal while the invoice is in `draft` — once it
 * is finalized, void it and create a new one instead so the audit trail stays
 * intact. Only the fields you pass are changed; replacing `lineItems`
 * recomputes the totals server-side.
 *
 * The SDK method is `update()` (`PATCH /v1/invoices/{id}`); "revise" is the
 * domain term for editing a draft.
 *
 * Run:
 *   npx tsx examples/invoices/update.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const revised = await client.invoices.update('inv_replace_with_real_id', {
  notes: 'Net 30. Updated per customer request.',
  dueAt: '2026-07-01T00:00:00.000Z',
  lineItems: [
    { description: 'Consulting — May 2026', quantity: 10, unitAmount: 12_500 }, // bumped 8 → 10
  ],
})

console.log(`revised ${revised.id ?? '?'} [${revised.status ?? '?'}]`)
console.log(`  subtotal ${String(revised.subtotal ?? 0)}, total ${String(revised.total ?? 0)} USD`)

File: sdk-python/examples/invoices/update.py
"""Revise a draft invoice.

Only legal while the invoice is in ``draft`` — once finalized, void it and
create a new one instead so the audit trail stays intact. Replacing
``line_items`` recomputes the totals server-side; only the fields you pass are
changed. The method is ``update``; "revise" is the domain term for editing a
draft.

Run with: python examples/invoices/update.py
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.invoices import InvoiceLineItem, UpdateBody


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        revised = client.invoices.update(
            "inv_replace_with_real_id",
            UpdateBody(
                notes="Net 30. Updated per customer request.",
                due_at="2026-07-01T00:00:00.000Z",
                line_items=[
                    InvoiceLineItem(
                        description="Consulting (revised)",
                        quantity=10,
                        unit_amount=12_500,
                    ),
                ],
            ),
        )
        print(f"revised {revised.id} [{revised.status}]")
        print(f"  subtotal {revised.subtotal}, total {revised.total}")


if __name__ == "__main__":
    main()
